### ISSUE TITLE
Why is Linux unsupported? It seems to run fine

### ISSUE THREAD

**Issue by Schmoho** [emmamglass/reconstructor#8]
Hello,

It seems to me that the only OS-dependent code in this project deals with OS-paths, which under Linux is basically the same as under Darwin. So I just modified the code to behave the same for Linux and Darwin and ran the test suite and it finished without errors.

Am I missing something here? Why is Linux not supported?

Best Regards

**Comment by abates20** [comment]
Really the only thing in Reconstructor that is OS-dependent is using DIAMOND to perform the BLAST search. Unfortunately, the installation of DIAMOND is different between MacOS and Linux, so simply changing the code to behave the same for Linux and Darwin won't guarantee that it works for Linux. Adding support would require at least 2 things I can think of:

1. Either put the linux executable for DIAMOND in reconstructor (like we do for Windows) or add functionality in reconstructor's test suite to grab and unzip the linux executable from DIAMOND's GitHub page. I favor the latter option and would like to also do this for Windows so that the Window's executable is not included in all distributions regardless of operating system.
2. Update the code in reconstructor that calls DIAMOND to handle calling the Linux executable.

All that being said, adding support for Linux is definitely on my radar, but I am planning to do a pretty significant refactoring of the reconstructor code first to make it more maintainable long term.

**Comment by drboothtj** [comment]
Hi,

I just wanted to follow up. Is this still the process for getting Linux functionality or was it updated?

I installed with pip but then failed the tests with an error saying Linux is not supported.

Any advice on what to do next would be gratefully received! 

Thanks!

**Comment by drboothtj** [comment]
Thanks for the reply and, by the looks of it, the update!

I will give the new version a shot!

**Comment by abates20** [comment]
Yes, with the update to version 1.2.0, Reconstructor should now work on Linux! If you run into any problems with it, feel free to open a new issue.
